### PR TITLE
DB-2372: fixing system addon update request and enable flag

### PR DESCRIPTION
**Pull request by tsl143**

### CODE REVIEW

File: mozilla-release/toolkit/mozapps/extensions/internal/XPIInstall.jsm [hunk 1]
```diff
@@ -4012,6 +4012,11 @@ var XPIInstall = {
   },
 
   async updateSystemAddons() {
+    const PREF_SYS_ADDON_UPDATE_ENABLED = "extensions.systemAddon.update.enabled";
+    if (!Services.prefs.getBoolPref(PREF_SYS_ADDON_UPDATE_ENABLED, true)) {
+      return false;
```
Comment (remusao) [inline mozilla-release/toolkit/mozapps/extensions/internal/XPIInstall.jsm]: This method does not seem to expect any return value.
```suggestion
      return;
```

File: mozilla-release/toolkit/mozapps/update/UpdateService.jsm [hunk 2]
```diff
@@ -3941,6 +3941,10 @@ Checker.prototype = {
       return;
     }
 
+    // CLIQZ-SPECIAL Start checking updates for System Addons also.
+    let { XPIProvider } = Components.utils.import("resource://gre/modules/addons/XPIProvider.jsm", {});
```
Comment (remusao) [inline mozilla-release/toolkit/mozapps/update/UpdateService.jsm]: Nit.
```suggestion
    const { XPIProvider } = Components.utils.import("resource://gre/modules/addons/XPIProvider.jsm", {});
```

Comment (tsl143) [reply mozilla-release/toolkit/mozapps/update/UpdateService.jsm]: Firefox uses `const` as DEFINE only.

Comment (remusao) [reply mozilla-release/toolkit/mozapps/update/UpdateService.jsm]: #OldSchool :smile:

**Pull request merged**